PochtaBot.py

```diff
@@ -0,0 +1,274 @@
+                if buttons_counter > CFG_MAX_BUTTONS_TO_SHOW:
+                    break
+
+                item = types.KeyboardButton(non_delivered_shipment.barcode)
+                markup.add(item)
+                buttons_counter += 1
+
+    except Exception as e:
+        print("Exception in draw buttons procedure", e)
+        if DEBUG:
+            raise e
+
+    # show Start/Stop Automated Notification button
+    item = types.KeyboardButton(
+        COMMAND_STOP_NOTIFICATION if chat_id in auto_notified_chats else COMMAND_START_NOTIFICATION
+    )
+    markup.add(item)
+    return markup
+
+
+# Handle user messages
+@bot.message_handler(content_types=["text"])
+def handle_text(message):
+    try:
+        user_entry = message.text.strip()
+        answer = None
+        with db_session:
+
+            if user_entry == COMMAND_LIST_READY_FOR_COLLECTION:
+
+                # we will find shipments that are registered in DB but not marked as arrived/delivered
+                shipments_to_check_for_arrival = select(
+                    shipment for shipment in Shipment if shipment.chat == message.chat.id and \
+                    (shipment.last_event != 2 and not (shipment.last_event == 8 and shipment.last_event_result == 2))
+                )
+                # and will request Russian Post for their up-to-date status and commit this updated status to DB
+                for shipment_db_record in shipments_to_check_for_arrival:
+                    shipment_xml = RussianPostAPI.get_shipment_data(
+                        shipment_db_record.barcode, CFG_POSTAL_API_KEY, CFG_POSTAL_API_PASS
+                    )
+                    shipment_info = ShipmentInfoParser.parse_xml(shipment_xml)
+                    shipment_db_record.last_event = shipment_info.events[-1][0]
+                    shipment_db_record.last_event_result = shipment_info.events[-1][4]
+
+                # now we have uo-to-date status of arrival in the DB
+                # reading it from the DB
+                arrived_shipments = select(
+                    shipment for shipment in Shipment if shipment.chat == message.chat.id and shipment.last_event == 8 \
+                    and shipment.last_event_result == 2
+                )
+
+                answer = ""
+                for shipment_db_record in arrived_shipments:
+                    answer += str(shipment_db_record.barcode) + "\n"
+                if not answer:
+                    answer = MESSAGE_NOTHING_READY_FOR_COLLECTION
+
+            elif user_entry == COMMAND_START_NOTIFICATION:
+                auto_notified_chats.add(message.chat.id)
+                answer = BUTTON_AUTO_NOTIFICATION_ON
+
+            elif user_entry == COMMAND_STOP_NOTIFICATION:
+                if message.chat.id in auto_notified_chats:
+                    auto_notified_chats.remove(message.chat.id)
+                answer = BUTTON_AUTO_NOTIFICATION_OFF
+
+            elif user_entry and 13 <= len(user_entry) <= 14:
```

Comment:
As for me, it's not obvious how it would work without braces. It's up to you, but maybe it's better to add them

Reply:
added braces and a comment explaining what the constants 13 and 14 mean